src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/CO2Solubility.cpp

```diff
@@ -321,8 +322,28 @@ CO2Solubility::CO2Solubility( string const & name,
   string const expectedWaterPhaseNames[] = { "Water", "water", "Liquid", "liquid" };
   m_phaseLiquidIndex = PVTFunctionHelpers::findName( phaseNames, expectedWaterPhaseNames, "phaseNames" );
 
-  m_CO2SolubilityTable = makeSolubilityTable( inputParams, m_modelName, FunctionManager::getInstance() );
-  m_WaterVapourisationTable = makeVapourisationTable( inputParams, m_modelName, FunctionManager::getInstance() );
+  string const solubilityModel = 10 < inputParams.size() ? inputParams[10] : "DuanSun";
+  FunctionManager & functionManager = FunctionManager::getInstance();
+
+  if( solubilityModel == "SpycherPruess" )
+  {
+    std::pair< TableFunction const *, TableFunction const * > tables =
+      CO2SolubilitySpycherPruess::makeSolubilityTables( inputParams, m_modelName, functionManager );
+    m_CO2SolubilityTable = tables.first;
+    m_WaterVapourisationTable = tables.second;
+  }
+  else if( solubilityModel == "DuanSun" )
+  {
+    m_CO2SolubilityTable = makeSolubilityTable( inputParams, m_modelName, functionManager );
```

Comment:
Wouldn't it be better to have `CO2Solubility` to be pure virtual and each model in a class, hence `CO2SolubilitySpycherPruess` and `CO2SolubilityDuanSun`